patches/net/minecraft/server/network/ServerGamePacketListenerImpl.java.patch

```diff
@@ -17,14 +17,37 @@
              }
  
              entity.absMoveTo(d3, d4, d5, f, f1);
-+            this.player.absMoveTo(d3, d4, d5, this.player.getYRot(), this.player.getXRot()); // Forge - Resync player position on vehicle moving
++            resyncPlayerWithVehicle(entity); // Neo - Resync player position on vehicle moving
              boolean flag3 = serverlevel.noCollision(entity, entity.getBoundingBox().deflate(0.0625));
              if (flag && (flag2 || !flag3)) {
                 entity.absMoveTo(d0, d1, d2, f, f1);
-+               this.player.absMoveTo(d3, d4, d5, this.player.getYRot(), this.player.getXRot()); // Forge - Resync player position on vehicle moving
++               resyncPlayerWithVehicle(entity); // Neo - Resync player position on vehicle moving
                 this.send(new ClientboundMoveVehiclePacket(entity));
                 return;
              }
+@@ -443,6 +_,22 @@
+       }
+    }
+ 
++   private void resyncPlayerWithVehicle(Entity vehicle) {
++      Vec3 oldPos = this.player.position();
++      float yRot = this.player.getYRot();
++      float xRot = this.player.getXRot();
++      float yHeadRot = this.player.getYHeadRot();
++      vehicle.positionRider(this.player);
++      // keep old rotation
++      this.player.setYRot(yRot);
++      this.player.setXRot(xRot);
++      this.player.setYHeadRot(yHeadRot);
```

Comment:
I'm note sure how to test it, but I have the slight concern that this would override the player's rotation (as the rider of the vehicle) which was set by the `positionRider` call. My concern is born from looking at the code in `Boat#positionRider`, which clamps the rotation of the rider to align with the boat's.

Reply:
As I mentioned in the PR description, this is intended as the original patch also preserved rotation.

Reply:
Hmmmm. I'll let it through, but someone (in the future) should probably investigate that and determine exactly why preserving rotation is needed, to either add as an explanatory comment or make the patch obsolete (if it turns out its been fixed in the meantime).